### ISSUE TITLE
AdvancedCollection inserts element at wrong position when the element is added at the end of the source collection

### ISSUE THREAD

**Issue by abdes** [CommunityToolkit/Windows#317]
### Describe the bug

In [AdvancedCollection.cs](https://github.com/CommunityToolkit/Windows/blob/2735c8718984daa2ddee7a5aa9235f57bead0df7/components/Collections/src/AdvancedCollectionView/AdvancedCollectionView.cs#L699), 

```csharp
        private bool HandleItemAdded(int newStartingIndex, object newItem, int? viewIndex = null)
        {
...
                else if (newStartingIndex == _source.Count - 1)
                {
                    newViewIndex = _view.Count - 1;
                }
...
}
```

When an item is added at the end of the source collection (i.e. `_source.Count -1`), the AdvancedCollection will insert it at the position `_view.Count - 1`, which is the position just before the last item in the view. This is not correct and results in the items order not respecting the source collection order, if the view does not use sorting.

### Steps to reproduce

```text
Please refer to the source code above, it's obvious.
```

### Expected behavior

The expected behavior is that the item is added at the end of the collection.

### Code Platform

- [ ] UWP
- [ ] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 10 21H2 (Build 19044)
- [ ] Windows 10 22H2 (Build 19045)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Windows 10, version 2104 (Build 20348)
- [ ] Windows 11, version 22H2 (Build 22000)
- [ ] Other (specify)

### Help us help you

Yes, I'd like to be assigned to work on this item.

**Comment by abdes** [comment]
Also submitted as: https://github.com/CommunityToolkit/WindowsCommunityToolkit/issues/4965

**Comment by zotya701** [comment]
Just run to this problem the other day. Had to use sorting, which decreases the performance a little bit.

**Comment by michael-hawker** [comment]
Thanks @abdes, I believe someone had tried to submit a fix for this before: https://github.com/CommunityToolkit/WindowsCommunityToolkit/pull/4814

Related to #307 and the #309 PR.

However, we really need proper tests here and to ensure we have the correct and desired behavior for various scenarios. So we need ones that show the issue, and also other similar edge cases, and that the submitted fix resolves the broken scenario without impacting the others.

**Comment by Arlodotexe** [comment]
> #309 it is, but not really #307.

@abdes PR #309 closes issue #307. Closing this as a duplicate of #307.
